Tools/speech_recognizer/GNUmakefile

```diff
@@ -0,0 +1,36 @@
+include $(GNUSTEP_MAKEFILES)/common.make
+
+# Set the default.  Eventually this should be conditional.
+GSSPEECHRECOGNITIONENGINE = Pocketsphinx
+
+VERSION = 0.1
+
+# This is an app not a tool because, eventually, it will present the user
+# interface for the GUI part of the speech engine.
+APP_NAME = GSSpeechRecognitionServer
+
+GSSpeechRecognitionServer_LANGUAGES = English
+
+GSSpeechRecognitionServer_OBJC_FILES = \
+	GSSpeechRecognitionEngine.m \
+	GSSpeechRecognitionServer.m \
+	GSSpeechRecognizer.m \
+	main.m
+
+
+ifeq ($(GSSPEECHRECOGNITIONENGINE), Pocketsphinx)
+POCKETSPHINX_BASE_LIBS = -lpocketsphinx -lsphinxbase -lsphinxad
+
+# Add includes and link dirs.... need to replace triplet.
+GSSpeechRecognitionServer_OBJC_FILES += PocketsphinxSpeechRecognitionEngine.m
+GSSpeechRecognitionServer_INCLUDE_DIRS += -I/usr/local/include/pocketsphinx \
```

Comment:
Isn't there some sort of autodetection for Pocketsphinx? This way we have quite a lot of hard coded paths that won't work on most machines.

Reply:
Agreed I will find a better way to do this.

Reply:
Why not use pkg-config as described here https://cmusphinx.github.io/wiki/tutorialpocketsphinx/

Reply:
> I am very unsure whether the way the work gets split up between the NSSpeechRecognizer class and its subclass GSSpeechRecognizer that lives in a separate process is efficient. I would prefer to have Richard look at that code as well. He has a lot more experience with server processes.

I understand.   I modeled this very much after the speech synthesis component.  If we can improve how it's structured it might also be worth reviewing that side as well.

Reply:
Done.